R/rvar-.R

```diff
@@ -90,7 +150,11 @@
 #' x
 #'
 #' @export
-rvar <- function(x = double(), dim = NULL, dimnames = NULL, nchains = NULL, with_chains = FALSE) {
+rvar <- function(
+  x = double(), dim = NULL, dimnames = NULL,
+  nchains = NULL, with_chains = FALSE,
+  log_weights = NULL
```

Comment:
I think `log_weights` needs input checking to ensure it is a vector. Currently it is possible to provide `log_weights` as a matrix which then causes issues when computing e.g. `mean` of the weighted `rvar`

Reply:
Good point. 

This should be true of weights everywhere right? `new_rvar()` calls down to `validate_weights()` so then something like `assert_atomic_vector(weights)` should go there. I can add that.